tests/contrib/aioredis/test_aioredis.py

```diff
@@ -0,0 +1,150 @@
+        We wrap the correct methods
+    When unpatching aioredis library
+        We unwrap the correct methods
+    """
+    if aioredis_version >= (2, 0):
+        assert isinstance(aioredis.client.Redis.execute_command, ObjectProxy)
+        assert isinstance(aioredis.client.Redis.pipeline, ObjectProxy)
+        assert isinstance(aioredis.client.Pipeline.pipeline, ObjectProxy)
+        unpatch()
+        assert not isinstance(aioredis.client.Redis.execute_command, ObjectProxy)
+        assert not isinstance(aioredis.client.Redis.pipeline, ObjectProxy)
+        assert not isinstance(aioredis.client.Pipeline.pipeline, ObjectProxy)
+    else:
+        assert isinstance(aioredis.Redis.execute, ObjectProxy)
+        unpatch()
+        assert not isinstance(aioredis.Redis.execute, ObjectProxy)
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_basic_request(redis_client):
+    val = await redis_client.get("cheese")
+    assert val is None
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_long_command(redis_client):
+    length = 1000
+    val_list = await redis_client.mget(*range(length))
+    assert len(val_list) == length
+    for val in val_list:
+        assert val is None
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_override_service_name(redis_client):
+    with override_config("aioredis", dict(service_name="myaioredis")):
+        val = await redis_client.get("cheese")
+        assert val is None
+        await redis_client.set("cheese", "my-cheese")
+        val = await redis_client.get("cheese")
+        if isinstance(val, bytes):
+            val = val.decode()
+        assert val == "my-cheese"
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_pin(redis_client):
+    Pin.override(redis_client, service="my-aioredis")
+    val = await redis_client.get("cheese")
+    assert val is None
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot(variants={"": aioredis_version >= (2, 0), "13": aioredis_version < (2, 0)})
+async def test_pipeline_traced(redis_client):
+    if aioredis_version >= (2, 0):
+        p = await redis_client.pipeline(transaction=False)
+        await p.set("blah", "boo")
+        await p.set("foo", "bar")
+        await p.get("blah")
+        await p.get("foo")
+    else:
+        p = redis_client.pipeline()
+        p.set("blah", "boo")
+        p.set("foo", "bar")
+        p.get("blah")
+        p.get("foo")
+
+    response_list = await p.execute()
+    assert response_list[0] is True  # response from redis.set is OK if successfully pushed
+    assert response_list[1] is True
+    assert (
+        response_list[2].decode() == "boo"
+    )  # response from hset is 'Integer reply: The number of fields that were added.'
+    assert response_list[3].decode() == "bar"
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot(variants={"": aioredis_version >= (2, 0), "13": aioredis_version < (2, 0)})
+async def test_two_traced_pipelines(redis_client):
```

Comment:
I think we should wrap all the pipeline operations with a parent span and ensure that the parenting is correct.

I didn't think about having two separate pipelines but that's a good addition! Otherwise the test looks good.